watertap/unit_models/tests/test_surrogate_crystallizer.py

```diff
@@ -0,0 +1,229 @@
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import os
+import pytest
+from pyomo.environ import (
+    ConcreteModel,
+    Var,
+    Constraint,
+    units as pyunits,
+    assert_optimal_termination,
+)
+from pyomo.util.check_units import assert_units_consistent
+from idaes.core import FlowsheetBlock
+from idaes.core.util.exceptions import ConfigurationError
+from watertap.property_models.water_prop_pack import WaterParameterBlock
+from watertap.property_models.multicomp_aq_sol_prop_pack import (
+    MCASParameterBlock,
+    MaterialFlowBasis,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+from idaes.core.surrogate.surrogate_block import SurrogateBlock
+from idaes.core.surrogate.pysmo_surrogate import (
+    PysmoSurrogate,
+)
+from watertap.unit_models.surrogate_crystallizer import SurrogateCrystallizer
+
+from idaes.core import UnitModelCostingBlock
+from watertap.costing import WaterTAPCosting
+from watertap.core.solvers import get_solver
+
+__author__ = "Oluwamayowa Amusat, Adam Atia"
+
+
+# TODO: can consider adding NN model surrogate function and test
+def add_crystallizer_rbf_model(
+    blk,
+    surrogate_inputs_with_bounds,
+    surrogate_outputs,
+    surrogate_to_flowsheet_basis_ratio,
+):
+    ##############################################################################################################
+    # block loading into IDAES Surrogate Block
+    ##############################################################################################################
+    crystallizer_inputs = [
+        surrogate_inputs_with_bounds[k]["flowsheet_var"]
+        for k in surrogate_inputs_with_bounds.keys()
+    ]
+    crystallizer_outputs = [
+        surrogate_outputs[k]["flowsheet_var"] for k in surrogate_outputs.keys()
+    ]
+
+    try:
+        assert set(list(blk.solids_list)).issubset(surrogate_outputs.keys())
+    except AssertionError:
+        raise ConfigurationError(
+            "List of solids provided in crystallizer model must match surrogate output keys for solids. Please check."
+        )
+
+    filename = [
+        "Vapor_Pressure",
+        "Calcite_g",
+        "Anhydrite_g",
+        "Glauberite_g",
+        "Halite_g",
+    ]
+    surrogate_directory = os.path.dirname(os.path.abspath(__file__))
+    surrogate_directory = os.path.join(
+        surrogate_directory, "surrogate_crystallizer_defaults"
+    )
```

Comment:
```suggestion
    surrogate_directory = Path(this_file_dir())
```

Reply:
This change might address the failing checks.

Reply:
I'm not sure it will since the checks fail for user mode, if the user installed through pip. I have moved the surrogates to the data folder and am trying another way.